hack/template-lint.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -x
```

Comment:
Maybe drop the `-x`?  I checked this out and bumped a file to run this on an existing enhancement, and:

```console
 hack/template-lint.sh 
++ dirname hack/template-lint.sh
+ TEMPLATE=hack/../guidelines/enhancement_template.md
++ sort -u
++ grep '^enhancements'
++ grep '\.md$'
++ git log --name-only --pretty= master..
+ CHANGED_FILES=enhancements/update/cluster-profiles.md
+ RC=0
+ for file in $CHANGED_FILES
+ grep '^##' hack/../guidelines/enhancement_template.md
+ read header_line
+ grep -v '\[optional\]'
+ grep -q '^## Release Signoff Checklist' enhancements/update/cluster-profiles.md
+ read header_line
+ grep -q '^## Summary' enhancements/update/cluster-profiles.md
+ read header_line
+ grep -q '^## Motivation' enhancements/update/cluster-profiles.md
...
```

is a lot noisier than:

```console
$ hack/template-lint.sh 
enhancements/update/cluster-profiles.md missing "### Risks and Mitigations"
enhancements/update/cluster-profiles.md missing "## Design Details"
enhancements/update/cluster-profiles.md missing "### Test Plan"
enhancements/update/cluster-profiles.md missing "### Graduation Criteria"
enhancements/update/cluster-profiles.md missing "#### Dev Preview -> Tech Preview"
enhancements/update/cluster-profiles.md missing "#### Tech Preview -> GA"
enhancements/update/cluster-profiles.md missing "#### Removing a deprecated feature"
enhancements/update/cluster-profiles.md missing "### Upgrade / Downgrade Strategy"
enhancements/update/cluster-profiles.md missing "### Version Skew Strategy"
enhancements/update/cluster-profiles.md missing "## Drawbacks"
```

Reply:
D'oh, I left that in from my last test run. It is removed now.